modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/tasks/RepairRequestTask.java

```diff
@@ -458,6 +479,9 @@ public RepairEntryProcessor(
 
                 GridCacheContext cctx = (GridCacheContext)entry.unwrap(GridCacheContext.class);
 
+                if (!cctx.affinity().affinityTopologyVersion().equals(startTopVer))
```

Comment:
What if client topology event has happened?
Comparing cctx.exchange().lastAffinityChangedTopologyVersion(<currentTopologyVersion>) with startTopVer should help.

Reply:
Fixed.